examples/point_transformer_segmentation.py

```diff
@@ -0,0 +1,250 @@
+from point_transformer_classification import TransitionDown, TransformerBlock
+
+import os.path as osp
+
+import torch
+import torch.nn.functional as F
+from torch.nn import Sequential as Seq, Linear as Lin, BatchNorm1d as BN, ReLU
+
+import torch_geometric.transforms as T
+from torch_geometric.datasets import ShapeNet
+from torch_geometric.loader import DataLoader
+from torch_geometric.utils import intersection_and_union as i_and_u
+from torch_geometric.nn.unpool import knn_interpolate
+
+from torch_cluster import knn_graph
+
+category = 'Airplane'  # Pass in `None` to train on all categories.
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'ShapeNet')
+transform = T.Compose([
+    T.RandomTranslate(0.01),
+    T.RandomRotate(15, axis=0),
+    T.RandomRotate(15, axis=1),
+    T.RandomRotate(15, axis=2),
+])
+pre_transform = T.NormalizeScale()
+train_dataset = ShapeNet(path, category, split='trainval', transform=transform,
+                         pre_transform=pre_transform)
+test_dataset = ShapeNet(path, category, split='test',
+                        pre_transform=pre_transform)
+train_loader = DataLoader(train_dataset, batch_size=10, shuffle=True)
+test_loader = DataLoader(test_dataset, batch_size=10, shuffle=False)
+
+
+class TransitionUp(torch.nn.Module):
+    '''
+        Reduce features dimensionnality and interpolate back to higher
+        resolution and cardinality
+    '''
+
+    def __init__(self, in_channels, out_channels):
+        super(TransitionUp, self).__init__()
+        self.mlp_sub = Seq(
+            Lin(in_channels, out_channels),
+            BN(out_channels),
+            ReLU()
+        )
+
+        self.mlp = Seq(
+            Lin(out_channels, out_channels),
+            BN(out_channels),
+            ReLU()
+        )
+
+    def forward(self, x, x_sub, pos, pos_sub, batch=None, batch_sub=None):
+        # transform low-res features and reduce the number of features
+        x_sub = self.mlp_sub(x_sub)
+
+        # interpolate low-res feats to high-res points
+        x_interpolated = knn_interpolate(
+            x_sub, pos_sub, pos, k=3, batch_x=batch_sub, batch_y=batch)
+
+        x = self.mlp(x) + x_interpolated
+
+        return x, pos
```

Comment:
Do we return `pos` just for convenience (as it is simply passed through)?

Reply:
Indeed, I didn't realize it. I changed it and pos is not returned anymore along with x